Issue zonemaster engine 171

vlevigneron opened the pull request:
Add the new step in the algorithm.
MIL and ET still do not pass, but VU and MP do.

matsduf commented on lib/Zonemaster/Test/Nameserver.pm:
```diff
@@ -216,6 +216,9 @@ sub nameserver01 {
+            elsif ( not $p->is_redirect and not $p->aa and not $p->answer and $p->rcode eq q{NOERROR} ) {
+                $is_not_recursor{ $local_ns->address->short }++;
+            }
             $nsnames{ $local_ns->name }++;
```
@vlevigneron, can you add a comment that states what use case that the new code will capture?

matsduf commented on lib/Zonemaster/Test/Nameserver.pm:
```diff
@@ -272,6 +275,9 @@ sub nameserver01 {
+                    elsif ( not $p->is_redirect and not $p->aa and not $p->answer and $p->rcode eq q{NOERROR} ) {
+                        $is_not_recursor{ $local_ns->address->short }++;
+                    }
                     $nsnames{ $local_ns->name }++;
```
@vlevigneron, same thing here, can you add a comment that states what use case that the new code will capture?

vlevigneron commented:
Well, I added that test because you asked for in a preview message of this
tread. You wrote:

"

The ET server has the same problem as MIL.

The MP servers returns answers with AA unset, ANSWER==0 and status==NOERROR
but no referal. That should also be OK.

One VU server is like the MP servers.

Let us update algorithm to make sure that MP and VU passes, but I see no
algorithm that would let MIL and ET pass.
"
That's what I did. I just added the MP case taht makes MP and VU passes.

2016-11-23 12:16 GMT+01:00 matsduf <notifications@github.com>:

> *@matsduf* commented on this pull request.
> ------------------------------
>
> In lib/Zonemaster/Test/Nameserver.pm
> <https://github.com/dotse/zonemaster-engine/pull/247#pullrequestreview-9841399>
> :
>
> > @@ -216,6 +216,9 @@ sub nameserver01 {
>              elsif ( $p->is_redirect and not $p->aa ) {
>                  $is_not_recursor{ $local_ns->address->short }++;
>              }
> +            elsif ( not $p->is_redirect and not $p->aa and not $p->answer and $p->rcode eq q{NOERROR} ) {
> +                $is_not_recursor{ $local_ns->address->short }++;
> +            }
>              $nsnames{ $local_ns->name }++;
>
> @vlevigneron <https://github.com/vlevigneron>, can you add a comment that
> states what use case that the new code will capture?
> ------------------------------
>
> In lib/Zonemaster/Test/Nameserver.pm
> <https://github.com/dotse/zonemaster-engine/pull/247#pullrequestreview-9841399>
> :
>
> > @@ -272,6 +275,9 @@ sub nameserver01 {
>                      elsif ( $p->is_redirect and not $p->aa ) {
>                          $is_not_recursor{ $local_ns->address->short }++;
>                      }
> +                    elsif ( not $p->is_redirect and not $p->aa and not $p->answer and $p->rcode eq q{NOERROR} ) {
> +                        $is_not_recursor{ $local_ns->address->short }++;
> +                    }
>                      $nsnames{ $local_ns->name }++;
>
> @vlevigneron <https://github.com/vlevigneron>, same thing here, can you
> add a comment that states what use case that the new code will capture?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dotse/zonemaster-engine/pull/247#pullrequestreview-9841399>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFPtOI4keInyrLMUSMvqKzT39frA1Mfeks5rBCB-gaJpZM4K6AXy>
> .
>

matsduf commented:
I wish I would find more documentation in the code...
